src/lib/phdi-building-blocks/phdi_building_blocks/schemas.py

```diff
@@ -0,0 +1,251 @@
+
+    return data
+
+
+def make_resource_type_table(
+    resource_type: str,
+    schema: dict,
+    output_path: pathlib.PosixPath,
+    output_format: Literal["parquet"],
+    credential_manager: AzureFhirserverCredentialManager,
+):
+    """
+    Given a FHIR resource type, schema, and FHIR server credential manager create a
+    table containing the field from resource type specified in the the schema.
+
+    :param resource_type: A FHIR resource type.
+    :param schema: A schema specifying the desired values by FHIR resource type.
+    :param credential_manager: A credential manager for
+    a FHIR server.
+    """
+
+    output_path.mkdir(parents=True, exist_ok=True)
+    output_file_name = output_path / f"{resource_type}.{output_format}"
+
+    query = f"/{resource_type}"
+    response = query_fhir_server(credential_manager, query)
+
+    additional_page = True
+    writer = None
+    while additional_page:
+        if response.status_code != 200:
+            log_fhir_server_error(response.status_code)
+            break
+
+        # Load queried data.
+        query_result = json.loads(response.content)
+        raw_schema_data = []
+
+        # Extract values specified by schema from each resource.
+        for resource in query_result["entry"]:
+            values_from_resource = apply_schema_to_resource(
+                resource["resource"], schema
+            )
+            if values_from_resource != {}:
+                raw_schema_data.append(values_from_resource)
+
+        # Write data to parquet
+        writer = write_schema_table(
+            raw_schema_data, output_file_name, output_format, writer
+        )
+
+        # Check for an additional page of query results.
+        for link in query_result.get("link"):
+            if link.get("relation") == "next":
+                next_page_url = link.get("url")
+                response = query_fhir_server(
+                    credential_manager, specific_url=next_page_url
+                )
+                break
+            else:
+                response = None
+
+        if response is None:
+            additional_page = False
+    if writer is not None:
+        writer.close()
+
+
+def generate_schema(
+    fhir_url: str,
+    schema_path: pathlib.PosixPath,
+    output_path: pathlib.PosixPath,
+    output_format: Literal["parquet"],
+):
+    """
+    Given the url for a FHIR server, the location of a schema file, and and output
+    directory generate the specified schema and store the tables in the desired
+    location.
+
+    :param fhir_url: URL to a FHIR server.
+    :schema_path: A path to the location of a YAML schema config file.
+    :output_path: A path to the directory where tables of the schema should be written.
+    :output_format: Specifies the file format of the tables to be generated.
+    """
+
+    schema = load_schema(schema_path)
+    schema_name = list(schema.keys())[0]
+    schema = schema[schema_name]
```

Comment:
Again, this is limiting us to a single table, which won't work with the actual schema we'll be using.

```python
>>> schema = {"my_table": {"Patient": {"Identifier": {}}}, "my_other_table": {"Practitioner": {"Identifier": {}}}}
>>> schema_name = list(schema.keys())[0]
>>> schema = schema[schema_name]
>>> schema
{'Patient': {'Identifier': {}}}
```